server/src/modules/rules/helpers/rule.comparator.service.ts

```diff
@@ -373,25 +375,25 @@ export class RuleComparatorService {
     this.workerData = [];
   }
 
-  private doRuleAction<T>(val1: T, val2: T, action: RulePossibility): boolean {
```

Comment:
>val1: T, val2: T

`val1` and `val2` were never always the same type (e.g. string[] being compared against string with action CONTAINS), so